src/Providers/Stripe/Customer/Service.php

```diff
@@ -64,54 +64,86 @@ public function find(string $uid): Customer
     /**
      * Associates a card to a customer using the Stripe SDK.
      *
-     * @param Customer $customer
-     * @param Card $card
+     * @param string $customerUid
+     * @param string $cardUid
      *
      * @return Card
      */
-    public function addCard(Customer $customer, Card $card): Card
+    public function addCard(string $customerUid, string $cardUid): Card
     {
-        /** @var PaymentMethod $paymentMethod */
-        $paymentMethod = PaymentMethod::retrieve($card->uid);
-
-        $response = $paymentMethod->attach(['customer' => $customer->uid]);
-
-        return new Card(
-            $response->id,
-            $response->card->last4,
-            $response->card->brand,
-            $response->card->funding,
-            $response->card->exp_month,
-            $response->card->exp_year
-        );
+        switch ((new StripeToken($cardUid))->type) {
+            case StripeToken::PAYMENT_METHOD:
+                /** @var PaymentMethod $paymentMethod */
+                $paymentMethod = PaymentMethod::retrieve($cardUid);
+
+                $response = $paymentMethod->attach(['customer' => $customerUid]);
+
+                return new Card(
```

Comment:
I think we are duplicating some code here, personally I would even go for `$response = $response->card`
Not to duplicate card creation.

Reply:
As discussed the id is not part of the card object. Also I tried to modify the response object and Stripe lib won't let you do it so the solution would be to create an array but it will add more code for the same result.